### PR TITLE
Feature/cvsb 11333

### PR DESCRIPTION
**Pull request by m-ujaffer**
https://jira.dvsacloud.uk/browse/CVSB-11333

### CODE REVIEW

File: src/functions/updateTechRecord.ts [hunk 1]
```diff
@@ -10,12 +10,17 @@ export function updateTechRecordStatus(event: SQSEvent) {
...
     event.Records.forEach((record: SQSRecord) => {
         const test = JSON.parse(record.body);
-        const promise = UpdateTechRecordService.updateStatusByVin(test.vin,
+        const promiseUpdateStatus = UpdateTechRecordService.updateStatusByVin(test.vin,
             test.testStatus,
             test.testTypes.testResult,
             test.testTypes.testTypeId,
             test.newStatus);
-        promisesArray.push(promise);
+        if (test.euVehicleCategory) {
```
Comment (agoncearenco) [inline src/functions/updateTechRecord.ts]: A submitted test-result will always have an `euVehicleCategory`. Won't this condition try to update a tech-record's `euVehicleCategory` even if it already present?

Comment (agoncearenco) [reply src/functions/updateTechRecord.ts]: Ah, I see, you do this check in tech-records

File: src/functions/updateTechRecord.ts [hunk 2]
```diff
@@ -10,12 +10,17 @@ export function updateTechRecordStatus(event: SQSEvent) {
...
+        const promiseUpdateEuCategory = UpdateTechRecordService.updateEuVehicleCategory(test.vin,
+                test.euVehicleCategory);
+        promisesArray.push(promiseUpdateEuCategory);
```
Comment (KHodderDVSA) [inline src/functions/updateTechRecord.ts]: indentation

File: src/services/UpdateTechRecordService.ts [hunk 3]
```diff
@@ -21,4 +21,20 @@ export class UpdateTechRecordService {
 
         return LambdaService.invoke(Configuration.getInstance().getEndpoints().functions.updateTechRecordStatus.name, event);
     }
+
+   public static updateEuVehicleCategory(vin: string, euVehicleCategory: string) {
+        const event = {
+            path: `/vehicles/${vin}/update-eu-vehicle-category`,
+            pathParameters: {
+                vin,
```
Comment (KHodderDVSA) [inline src/services/UpdateTechRecordService.ts]: Depending on when this gets merged, will need to be conscious of systemNumber change. Not a blocker at present.

**Pull request merged**